samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,144 @@
+"""
+Remote invoke executor implementation for Lambda
+"""
+import base64
+import json
+import logging
+from json import JSONDecodeError
+from typing import Any, Dict, cast
+
+from botocore.exceptions import ClientError, ParamValidationError
+from botocore.response import StreamingBody
+
+from samcli.lib.remote_invoke.exceptions import InvalidResourceBotoParameterException
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeExecutionInfo,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeRequestResponseMapper,
+)
+from samcli.lib.utils import boto_utils
+
+LOG = logging.getLogger(__name__)
+
+
+class LambdaInvokeExecutor(BotoActionExecutor):
+    """
+    Calls "invoke" method of "lambda" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _lambda_client: Any
+    _function_name: str
+
+    def __init__(self, lambda_client: Any, function_name: str):
+        self._lambda_client = lambda_client
+        self._function_name = function_name
+        self.request_parameters = {"InvocationType": "RequestResponse", "LogType": "Tail"}
+
+    def validate_action_parameters(self, parameters: dict) -> None:
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        :param parameters: Boto parameters provided as input
+        """
+        for parameter_key, parameter_value in parameters.items():
+            if parameter_key == "FunctionName":
+                LOG.warning("FunctionName is defined using the value provided for --resource-id option.")
+            elif parameter_key == "Payload":
+                LOG.warning(
+                    "Payload is defined using the value provided for either --payload or --payload-file options."
+                )
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+    def _execute_action(self, payload: str):
+        self.request_parameters["FunctionName"] = self._function_name
+        self.request_parameters["Payload"] = payload
+        LOG.debug(
+            "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        try:
+            response = self._lambda_client.invoke(**self.request_parameters)
+        except ParamValidationError as param_val_ex:
+            raise InvalidResourceBotoParameterException(
+                f"Invalid parameter key provided."
+                f" {str(param_val_ex).replace('FunctionName, ', '').replace('Payload, ', '')}"
+            ) from param_val_ex
+        except ClientError as client_ex:
+            validation_err_str = "ValidationException"
+            if boto_utils.get_client_error_code(client_ex) == validation_err_str:
```

Comment:
I would suggest to move `ValidationException` as a constant at the top, or use the string directly in the line below.